src/ray/gcs/gcs_server/gcs_table_storage.h

```diff
@@ -0,0 +1,373 @@
+ public:
+  explicit GcsTaskLeaseTable(
+      std::shared_ptr<StoreClient<TaskID, TaskLeaseData, JobID>> store_client)
+      : GcsTable(std::move(store_client)) {
+    table_name_ = TASK_LEASE_TABLE_NAME;
+  }
+};
+
+class GcsTaskReconstructionTable : public GcsTable<TaskID, TaskReconstructionData> {
+ public:
+  explicit GcsTaskReconstructionTable(
+      std::shared_ptr<StoreClient<TaskID, TaskReconstructionData, JobID>> store_client)
+      : GcsTable(std::move(store_client)) {
+    table_name_ = TASK_RECONSTRUCTION_TABLE_NAME;
+  }
+};
+
+class GcsObjectTable : public GcsTable<ObjectID, ObjectTableDataList> {
+ public:
+  explicit GcsObjectTable(
+      std::shared_ptr<StoreClient<ObjectID, ObjectTableDataList, JobID>> store_client)
+      : GcsTable(std::move(store_client)) {
+    table_name_ = OBJECT_TABLE_NAME;
+  }
+};
+
+class GcsNodeTable : public GcsTable<ClientID, GcsNodeInfo> {
+ public:
+  explicit GcsNodeTable(
+      std::shared_ptr<StoreClient<ClientID, GcsNodeInfo, JobID>> store_client)
+      : GcsTable(std::move(store_client)) {
+    table_name_ = NODE_TABLE_NAME;
+  }
+};
+
+class GcsNodeResourceTable : public GcsTable<ClientID, ResourceMap> {
+ public:
+  explicit GcsNodeResourceTable(
+      std::shared_ptr<StoreClient<ClientID, ResourceMap, JobID>> store_client)
+      : GcsTable(std::move(store_client)) {
+    table_name_ = NODE_RESOURCE_TABLE_NAME;
+  }
+};
+
+class GcsHeartbeatTable : public GcsTable<ClientID, HeartbeatTableData> {
+ public:
+  explicit GcsHeartbeatTable(
+      std::shared_ptr<StoreClient<ClientID, HeartbeatTableData, JobID>> store_client)
+      : GcsTable(std::move(store_client)) {
+    table_name_ = HEARTBEAT_TABLE_NAME;
+  }
+};
+
+class GcsHeartbeatBatchTable : public GcsTable<ClientID, HeartbeatBatchTableData> {
+ public:
+  explicit GcsHeartbeatBatchTable(
+      std::shared_ptr<StoreClient<ClientID, HeartbeatBatchTableData, JobID>> store_client)
+      : GcsTable(std::move(store_client)) {
+    table_name_ = HEARTBEAT_BATCH_TABLE_NAME;
+  }
+};
+
+class GcsErrorInfoTable : public GcsTable<JobID, ErrorTableData> {
+ public:
+  explicit GcsErrorInfoTable(
+      std::shared_ptr<StoreClient<JobID, ErrorTableData, JobID>> store_client)
+      : GcsTable(std::move(store_client)) {
+    table_name_ = ERROR_INFO_TABLE_NAME;
+  }
+};
+
+class GcsProfileTable : public GcsTable<UniqueID, ProfileTableData> {
+ public:
+  explicit GcsProfileTable(
+      std::shared_ptr<StoreClient<UniqueID, ProfileTableData, JobID>> store_client)
+      : GcsTable(std::move(store_client)) {
+    table_name_ = PROFILE_TABLE_NAME;
+  }
+};
+
+class GcsWorkerFailureTable : public GcsTable<WorkerID, WorkerFailureData> {
+ public:
+  explicit GcsWorkerFailureTable(
+      std::shared_ptr<StoreClient<WorkerID, WorkerFailureData, JobID>> store_client)
+      : GcsTable(std::move(store_client)) {
+    table_name_ = WORKER_FAILURE_TABLE_NAME;
+  }
+};
+
+class GcsTableStorage {
```

Comment:
This class shouldn't explicitly depends on Redis. 
I think we can make this `GcsTableStorage` a pure interface, and add `RedisGcsTableStorage` as a sub class.

Reply:
fixed